android/core/src/main/java/com/stadiamaps/ferrostar/core/FerrostarCore.kt

```diff
@@ -375,6 +400,9 @@ class FerrostarCore(
 
         NavigationState(tripState = newState, currentValue.routeGeometry, isCalculatingNewRoute)
       }
+
+      // Update the notification
+      foregroundServiceManager.onNavigationState(_state.value)
```

Comment:
This is super minor, but is there a reason we can't put this in `handleStateUpdate`? It seems to fit better over there.